services/document_conversion/v1.js

```diff
@@ -0,0 +1,120 @@
+module.exports = function(RED) {
+
+	var http = require('http');
+	var https = require('https');
+	
+	var cfEnv = require("cfenv");
+	var util = require("util");
+		
+	var watson = require('watson-developer-cloud');
+	var temp = require('temp');
+	temp.track();
+	var fs = require('fs');
+	var fileType = require('file-type');
+	var request = require('request');
+
+	var appEnv	 = cfEnv.getAppEnv();
+
+	var converts = [];
+		
+	// GNF: Read credentials information from the VCAP environment variable
+	for (var i in appEnv.services) {
+		// filter the services to include only the Convert ones
+		if (i.match(/^(document_conversion)/i)) {
+			converts = converts.concat(appEnv.services[i].map(function(v) {
+				return {
+					name: v.name, 
+					label: v.label, 
+					url: v.credentials.url,
+					username: v.credentials.username,
+					password: v.credentials.password
+					};
+			}));
+		}
+	}
+
+	util.log("*** CONVERT: "+JSON.stringify(converts));
+
+	// GNF: This method provides service credentials when prompted from the node editor
+	RED.httpAdmin.get('/convert/vcap', function(req,res) {
+		res.send(JSON.stringify(converts));
+	});
+
+	function ConvertNode(config) {
+
+		RED.nodes.createNode(this,config);
+
+		this.name = config.name;
+		this.username = config.username;
+		this.password = config.password;
+		this.service = config.service;		
+		this.target = config.target;		
+
+		var node = this;
+
+		this.doCall = function(msg) {
+			
+			var document_conversion = watson.document_conversion({
+				username: node.username,
+				password: node.password,
+				version: 'v1',
+				version_date: '2015-12-01'
+			});
+			
+			var stream_buffer = function (file, contents, cb) {
+				fs.writeFile(file, contents, function (err) {
+					if (err) throw err;
+					cb(fileType(contents).ext);
+				});
+			};
+
+			var stream_url = function (file, location, cb) {
+				var wstream = fs.createWriteStream(file);
+				wstream.on('finish', function () {
+					fs.readFile(file, function (err, buf) {
+						if (err) console.error(err);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'console'.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2392370589/issues/source?bid=3225115&fileBranchId=3225117#l75)